### PR TITLE
merge-styles api input argument typings relaxed (TypeScript 3.9 now compiles)

### PR DESCRIPTION
**Pull request by dzearing**
#### Pull request checklist

In testing TS 3.9, customers found compiler typing errors being thrown from merge-styles apis. With the typescript compiler checks getting more accuracy, this can happen.

This should resolve the issues while still maintaining type safety on inputs/outputs.

Note that `ObjectOnly` type is needed to ensure that falsely types do not get intersected with the non-falsey result.

Background on `ObjectOnly`:

Think about this case:

```tsx
const result = concatStyleSets({ a: { ... } } , undefined)
```

The `result` type would be:

```tsx
{ a: { ... } } & undefined
```

...Which results in `never`. Which is not quite what we're looking for. The `ObjectOnly` type ensures that the undefined does not turn the intersection into a `never` and returns at least an object definition in response.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/13165)

### CODE REVIEW

File: packages/merge-styles/src/IStyleSet.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export type __MapToFunctionType<T> = Extract<T, Function> extends never
  * It may optionally contain style functions for sub components in the special `subComponentStyles`
  * property.
  */
-export type IStyleSet<TStyleSet extends IStyleSet<TStyleSet>> = {
+export type IStyleSet<TStyleSet extends IStyleSet<TStyleSet> = any> = {
```
Comment (xugao) [inline packages/merge-styles/src/IStyleSet.ts:25]:
```suggestion
export type IStyleSet<TStyleSet extends IStyleSet> = {
```

Comment (xugao) [reply packages/merge-styles/src/IStyleSet.ts:25]: nvm this is not right

**Comment by dzearing** [comment]
Something doesn't seem right... :)

File: packages/merge-styles/src/IStyleSet.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export type __MapToFunctionType<T> = Extract<T, Function> extends never
  * It may optionally contain style functions for sub components in the special `subComponentStyles`
  * property.
  */
-export type IStyleSet<TStyleSet extends IStyleSet<TStyleSet>> = {
+export type IStyleSet<TStyleSet extends IStyleSet<TStyleSet> = { [key: string]: any }> = {
```
Comment (xugao) [inline packages/merge-styles/src/IStyleSet.ts:25]: why is this change needed?

File: packages/merge-styles/etc/merge-styles.api.md [hunk 3]
```diff
@@ -4,29 +4,28 @@
 
 ```ts
 
+// Warning: (ae-forgotten-export) The symbol "ObjectOnly" needs to be exported by the entry point index.d.ts
```
Comment (xugao) [inline packages/merge-styles/etc/merge-styles.api.md:7]: 1. can you add in PR description why this change is needed? what scenario was failing with which change made?

**Comment by xugao** [comment]
is the removal of `extends` the change that fixed the error in 3.9?

File: packages/merge-styles/etc/merge-styles.api.md (hunk 3 above)
Comment (dzearing) [reply packages/merge-styles/etc/merge-styles.api.md:7]: Think about this case:

```tsx
const result = concatStyleSets({ a: { ... } } , undefined)
```

What is the type of result? It would be:

```tsx
{ a: { ... } } & undefined
```

...Which results in `never`. Which is not quite what we're looking for. The `ObjectOnly` type ensures that the undefined does not turn the intersection into a `never` and returns at least an object definition in response.

File: packages/merge-styles/src/IStyleSet.ts (hunk 2 above)
Comment (dzearing) [reply packages/merge-styles/src/IStyleSet.ts:25]: Originally when defining all of the concat/merge style definitions, I had provided the default in all of the input use cases as`IStyleSet<any>`. That felt wrong for a number of reasons, and instead I moved the default initializer here (to control the default in one place.) Then I thought `any` was too liberal and scoped it to default to an object.

File: packages/merge-styles/etc/merge-styles.api.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dzearing) [reply packages/merge-styles/etc/merge-styles.api.md]: done

**Comment by ecraig12345** [comment]
@dzearing Can you check whether this fixes #13132?

**Comment by dzearing** [comment]
@ecraig12345 yup :)
